jazz_services-handler/index.js

```diff
@@ -20,584 +21,265 @@ module.exports.handler = (event, context, cb) => {
-														"ID": results.id,
-														"DESCRIPTION": description,
-														"REPOSITORY": repository,
-														"EMAIL": email,
-														"SLACKCHANNEL": slackChannel,
-														"TAGS": tags,
-														"ENDPOINTS": endpoint,
-														"STATUS": "active"
-													};
-
-													crud.update(inputs, function (err, results) {
-														if (err) {
-															HandleFailedEvents(sequenceNumber, err.error, payload, failureCodes.PR_ERROR_2.code);
-															logger.error(err.details);
-															return innerCallback({
-																"error": "unknown error updating service " + payload.SERVICE_NAME.S + " in service catalog."
-															});
-														} else {
-															HandleProcessedEvents(sequenceNumber, payload);
-															logger.info("updated service "  + payload.SERVICE_NAME.S + " in service catalog.");
-															return innerCallback(null, {
-																"message": "updated service "  + payload.SERVICE_NAME.S + " in service catalog."
-															});
-														}
-													});
-
-												}
-
-											});
-										}
-									}
+		}; //End of promise			
+	}
+	
+	function validateAuthToken(result){
+		return new Promise((resolve, reject) => {
+			if (result.statusCode === 200 && result.body && result.body.data) {
+				resolve(result.body.data.token);
+			}else{
+			   var err = new Error("Unauthorized " + result.statusCode);
+				err.error = result.error;
+				return reject(err);
+			}
+		}); //End of promise
+	}
 
-								} else {
-									//push un-interesting event to processed queue
-									processedEvents.push({
-										"sequence_id": sequenceNumber,
-										"failure_code": null,
-										"failure_message": null
-									});
-									innerCallback(null,processedEvents)
-								}
-							}catch(e){
-								HandleFailedEvents(failureCodes.PR_ERROR_2.type, JSON.stringify(e), results.payload, failureCodes.PR_ERROR_2.code);
-								innerCallback(e);
-							}
-						}]
-					}, function (err, results) {
-							callback(err);
-					});
-				}, function (err) {
-					if(err){
-						mainCallback(err)
-					} else{
-						mainCallback(null, {
-							"processed_events": processedEvents.length,
-							"failed_events": failedEvents.length
+	function handleError(errorType,message){
+		var err = new Error();
+		err.error = {};
+		err.error.errorType =  errorType;
+		err.error.message = message;
+		return err;
+	}
+	
+	function processRecord(configData, authToken){		
+		return new Promise((resolve, reject) => {			
+			for (var i=0; i< event.Records.length; i++){
+				var record = event.Records[i];
+				var sequenceNumber = record.kinesis.sequenceNumber;
+				var encodedPayload = record.kinesis.data;
+				checkInterest(encodedPayload,sequenceNumber)
+				.then(function(result) {
+					if(result.interested_event){
+						processEvent(result.payload,configData, authToken);
+						processedEvents.push({
```

Comment:
Since processEvent is a promise, return it from here and then have a *thenable* where you push into processedEvents depending on whether processEvent was resolved or rejected.